lib/internal/modules/esm/resolve.js

```diff
@@ -270,12 +270,17 @@ function finalizeResolution(resolved, base) {
       resolved.pathname, fileURLToPath(base), 'module');
   }
 
-  if (StringPrototypeEndsWith(resolved.pathname, '/')) return resolved;
   const path = fileURLToPath(resolved);
+  const stats = tryStatSync(path);
 
-  if (!tryStatSync(path).isFile()) {
-    throw new ERR_MODULE_NOT_FOUND(
+
+  if (!stats.isFile()) {
+    const err = new ERR_MODULE_NOT_FOUND(
       path || resolved.pathname, fileURLToPath(base), 'module');
+    if (stats.isDirectory()) {
+      err.directoryURL = resolved.toString();
```

Comment:
Not sure about using public property on the `Error` object, could also use a `Symbol`. @guybedford any thoughts?

Reply:
@aduh95 I think we went back to your original approach with the specific error code for the directory import as you had originally. My apologies for churn.